lambdas/services/document_manifest_job_service.py

```diff
@@ -0,0 +1,156 @@
+            for document in self.handle_duplicate_document_filenames()
+        }
+
+        job_id = self.write_zip_trace(documents_to_download)
+
+        return job_id
+
+    def filter_documents_by_reference(
+        self,
+        selected_document_references: list[str],
+    ) -> list[DocumentReference]:
+        logger.info("Filtering Selected Document References for manifest")
+
+        reference_set = set(selected_document_references)
+        matched_references = [
+            document_reference
+            for document_reference in self.documents
+            if document_reference.id in reference_set
+        ]
+
+        if not matched_references:
+            raise DocumentManifestJobServiceException(
+                400, LambdaError.ManifestFilterDocumentReferences
+            )
+
+        return matched_references
+
+    def handle_duplicate_document_filenames(self) -> list[DocumentReference]:
+        file_names_to_be_zipped = {}
+
+        for document in self.documents:
+            file_name = document.file_name
+            duplicated_filename = file_name in file_names_to_be_zipped
+
+            if duplicated_filename:
+                file_names_to_be_zipped[file_name] += 1
+                document.file_name = document.create_unique_filename(
+                    file_names_to_be_zipped[file_name]
+                )
+            else:
+                file_names_to_be_zipped[file_name] = 1
+
+        return self.documents
+
+    def write_zip_trace(
+        self,
+        documents_to_download: dict,
+    ) -> str:
+        logger.info("Writing Document Manifest zip trace to db")
+
+        zip_trace = DocumentManifestZipTrace(FilesToDownload=documents_to_download)
+        self.dynamo_service.create_item(
+            self.zip_trace_table, zip_trace.model_dump(by_alias=True)
+        )
+
+        return str(zip_trace.job_id)
+
+    def create_document_manifest_presigned_url(
+        self, job_id: str
+    ) -> DocumentManifestJob:
+        zip_trace = self.query_zip_trace(job_id=job_id)
+
+        match zip_trace.job_status:
+            case ZipTraceStatus.PENDING:
+                return DocumentManifestJob(jobStatus=ZipTraceStatus.PENDING, url="")
+            case ZipTraceStatus.PROCESSING:
+                return DocumentManifestJob(jobStatus=ZipTraceStatus.PROCESSING, url="")
+            case ZipTraceStatus.COMPLETED:
+                is_manifest_ready = self.s3_service.file_exist_on_s3(
+                    self.zip_output_bucket, zip_trace.zip_file_location
+                )
+                if not is_manifest_ready:
+                    raise DocumentManifestJobServiceException(
+                        404, LambdaError.ManifestMissingJob
+                    )
+                presigned_url = self.s3_service.create_download_presigned_url(
+                    s3_bucket_name=self.zip_output_bucket,
+                    file_key=zip_trace.zip_file_location,
```

Comment:
same issue as above, zip_trace.zip_file_location gave the whole s3 url, not the file_key, so it generated a wrong url